Players able to flag their own messages - Fixes #8005

cincospenguinos opened the pull request:
Fixes Issue #8005 - Players should be able to flag their own messages
### Changes

Players can now flag their own messages. Previously if a player sent a message in chat, they could not flag that message. Now all players can flag all messages that they personally have sent.

This is the previous front end behavior:
![old_flag](https://cloud.githubusercontent.com/assets/7362771/18476009/b842e010-7985-11e6-8bc4-94c1a006efb7.png)

This is the new front end behavior:
![flag_change](https://cloud.githubusercontent.com/assets/7362771/18476018/bb8d69ac-7985-11e6-9151-40eeb49d8036.png)

This is the front end behavior when you flag your own post:
![flag_myself](https://cloud.githubusercontent.com/assets/7362771/18476108/0a028748-7986-11e6-9c12-6660c4ac8f54.png)

---

UUID: 59197564-6563-4a98-9a35-7391033dd87a

Commit:
Players able to flag their own messages

crookedneighbor commented on website/server/controllers/api-v3/chat.js:
```diff
@@ -222,7 +222,8 @@ api.flagChat = {
 
-    if (message.uuid === user._id) throw new NotFound(res.t('messageGroupChatFlagOwnMessage'));
+    // Your own message should be flaggable
+    //if (message.uuid === user._id) throw new NotFound(res.t('messageGroupChatFlagOwnMessage'));
```
Please just remove this. Also, I bet the build will fail because there is a test that covers not being able to flag your own message.

The test should be updated to illustrate that you _can_ flag your own messages.

Also, the `messageGroupChatFlagOwnMessage` translation string should be removed.

``` bash
git grep "messageGroupChatFlagOwnMessage" common/locales/en
```

^ to find the file it is in

Commit:
Modify test case to match the new behavior

Commit:
Remove a comment and a locales message

cincospenguinos commented:
I removed what you gave me and modified the test case, although the test case just checks if the response is ok, not the contents of the response. Should I have it test the object returned instead?

crookedneighbor commented:
It's fine as is. It'll mostly serve as a reminder that being able to flag your own message is intentional.

Tier awarded! You can read more about [contributor tiers and rewards here](http://habitrpg.wikia.com/wiki/Contributor_Rewards).

Note that as your tier increases, it takes progressively more work to achieve the next one, so for the higher tiers, larger PRs or a few PRs are required. There's some information about that at Contributor Rewards, but feel free to ask here if you have questions!

cincospenguinos commented:
Cool! I'm glad I could help out.
